src/components/inputs/notes.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+    <div>
+        <textarea            
+            v-model="newNote"           
+            :id="$attrs.id"
+            :class="$attrs.class"
+            :required="$attrs.required" 
+            :placeholder="$attrs.placeholder"
+            :disabled="$attrs.disabled"
+            :rows="$attrs.rows" 
+            :cols="$attrs.cols" 
+            @change="save" >
+        </textarea>       
+   
+        <table class="table table-striped" v-if="notes?.length">
+            <tr v-for="(item, index) in notes" :key="index">
+                <td>{{item}}  </td>   
+                <td style="width:10px;"><span class="btn" @click="remove(index)"><i class="fa fa-trash-o icon"></i></span></td>
+            </tr>
+        </table>   
+    </div> 
+</template>
+
+<script setup>
+   import { ref, computed } from 'vue'
+   import {useUser} from '../../services/composables/useAuth'
+  
+   let lastNote = '';
+   const newNote = ref();
+   const user = useUser();
+
+   const model = defineModel({type:String, required:true});
+   
+   let notes = computed(()=>{
+        if(model.value)
+            return  (JSON.parse(model.value)).filter(e=>e!=lastNote);
+        return [];
+   }); 
+   
+   const remove=(index)=>{  
+        for (var i = 0; i < notes.value.length; i++) {        
+            if(notes.value[i] == notes.value[index]){             
+                notes.value.splice(i, 1);        
+            }
+        }  
+        save();  
+   };
+
+    const save=()=>{   
+        const timestamp = new Intl.DateTimeFormat('en-US', {
+            year: 'numeric',
+            month: 'short',
+            day: '2-digit',
+            hour: 'numeric',
+            minute: 'numeric',
+            second: 'numeric',
+            hour12: true
+        }).format(new Date());
+        const lNotes = notes.value;
+        if(lNotes[lNotes.length-1] == lastNote)
+             lNotes.splice(lNotes.length-1, 1);
```

Comment:
No need complexify the coding with lengh etc. 
simply us at() and pop().  at() with negative value start from the end of the array

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop

```suggestion
        if(lNotes.at(-1) == lastNote)
             lNotes.pop();
```

Reply:
done